SConstruct

```diff
@@ -339,6 +341,29 @@ if GetOption("clazy"):
 
 Export('env', 'qt_env', 'arch', 'real_arch')
 
+def exist_scan(node, env, path):
```

Comment:
what's this for?

Reply:
Default scanner of Sconstruct filters implicit dependencies by reading the content of the files by regexing "#include" terms. For instance, in the file body/board/inc/stm32f4xx.h, the default Scanner would just filtering based on the "#include" and therefore it would require the file stm32f405xx.h as a dependency. As the pictures below.
![image](https://github.com/commaai/openpilot/assets/47828508/7c387c0c-a0c4-40ff-b83d-e1c6fcba3497)
Also, some of the C standard files would also be required such as "stdint.h" which we generally don't care about since it rarely or never explicitedly edited, and we only care about the dependency files which are in relative folders